context/app/static/js/components/cells/CellExpressionHistogram/CellExpressionHistogram.jsx

```diff
@@ -26,8 +27,15 @@ function CellExpressionHistogram({ uuid, cellVariableName, isLoading, finishLoad
       setExpressionData(response.map((d) => d.values[cellVariableName]));
       finishLoading(loadingKey);
     }
-    fetchCellExpression();
-  }, [uuid, cellVariableName, finishLoading, loadingKey]);
+    if (loadedOnce.current) {
+      return;
+    }
+
+    if (isExpanded) {
+      fetchCellExpression();
```

Comment:
Sorry if I'm not getting it, but it feels like the goal is to avoid repeating an expensive process. Could `useMemo` be used here, instead of us keeping track for ourselves of what's already been done?